test/features/interceptor/authorization_interceptor_test.dart

```diff
@@ -0,0 +1,362 @@
+            requestOptions: RequestOptions(path: baseUrl),
+          ),
+          type: DioErrorType.badResponse,
+        );
+
+        authorizationInterceptors.setTokenAndAuthorityOidc(
+          newToken: OIDCFixtures.tokenOidcExpired,
+          newConfig: OIDCFixtures.oidcConfiguration);
+
+        when(authenticationClient.refreshingTokensOIDC(
+            OIDCFixtures.oidcConfiguration.clientId,
+            OIDCFixtures.oidcConfiguration.redirectUrl,
+            OIDCFixtures.oidcConfiguration.discoveryUrl,
+            OIDCFixtures.oidcConfiguration.scopes,
+            OIDCFixtures.tokenOidcExpired.refreshToken
+        )).thenAnswer((_) async => OIDCFixtures.tokenOidc);
+
+        when(accountCacheManager.getCurrentAccount()).thenAnswer((_) async => AccountFixtures.aliceAccount);
+
+        dioAdapter
+          ..onPost(
+            baseUrl,
+            (request) => request.throws(401, dioError401),
+          )
+          ..onPost(
+            baseUrl,
+            (request) => request.reply(200, {'message': 'Retry request successfully!'})
+          );
+        
+        final response = await dio.post(baseUrl);
+
+        expect(response.data, {'message': 'Retry request successfully!'});
+      });
+
+      test('The upload attachment request should not be blocked when returning onError in case the token expires', () async {
+        final dioError401 = DioError(
+          error: {'message': 'Token Expired'},
+          requestOptions: RequestOptions(
+              path: baseUrl,
+              extra: {
+                FileUploader.uploadAttachmentExtraKey: {
+                  'platform': 'mobile',
+                  'path': '/path_file/',
+                  'type': 'image/png',
+                  'size': 100,
+                }
+              }
+          ),
+          response: Response(
+            statusCode: 401,
+            requestOptions: RequestOptions(path: baseUrl),
+          ),
+          type: DioErrorType.badResponse,
+        );
+
+        authorizationInterceptors.setTokenAndAuthorityOidc(
+          newToken: OIDCFixtures.tokenOidcExpired,
+          newConfig: OIDCFixtures.oidcConfiguration);
+
+        when(authenticationClient.refreshingTokensOIDC(
+          OIDCFixtures.oidcConfiguration.clientId,
+          OIDCFixtures.oidcConfiguration.redirectUrl,
+          OIDCFixtures.oidcConfiguration.discoveryUrl,
+          OIDCFixtures.oidcConfiguration.scopes,
+          OIDCFixtures.tokenOidcExpired.refreshToken
+        )).thenAnswer((_) async => OIDCFixtures.tokenOidc);
+
+        when(accountCacheManager.getCurrentAccount()).thenAnswer((_) async => AccountFixtures.aliceAccount);
+
+        dioAdapter
+          ..onPost(
+            baseUrl,
+            (request) => request.throws(401, dioError401),
+          )
+          ..onPost(
+            baseUrl,
+            (request) => request.reply(200, {'message': 'Retry upload attachment request successfully!'})
+          );
```

Comment:
IMO, we can not test like this, because this `dioAdapter` only accept the last setup

Reply:
it means, we can not respond different body with the same route

Reply:
please try 
```
      test('The request should not be blocked when returning onError in case the token expires', () async {
        final dioError401 = DioError(
          error: {'message': 'Token Expired'},
          requestOptions: RequestOptions(path: baseUrl),
          response: Response(
            statusCode: 401,
            requestOptions: RequestOptions(path: baseUrl),
          ),
          type: DioErrorType.badResponse,
        );

        authorizationInterceptors.setTokenAndAuthorityOidc(
          newToken: OIDCFixtures.tokenOidcExpired,
          newConfig: OIDCFixtures.oidcConfiguration);

        dioAdapter.onPost(
          baseUrl,
          (request) => request.throws(401, dioError401),
        );

        when(authenticationClient.refreshingTokensOIDC(
            OIDCFixtures.oidcConfiguration.clientId,
            OIDCFixtures.oidcConfiguration.redirectUrl,
            OIDCFixtures.oidcConfiguration.discoveryUrl,
            OIDCFixtures.oidcConfiguration.scopes,
            OIDCFixtures.tokenOidcExpired.refreshToken
        )).thenAnswer((_) async {
          dioAdapter.onGet(
            baseUrl,
            (request) => request.reply(200, {'message': 'Retry request successfully!'})
          );
          return OIDCFixtures.tokenOidc;
        });

        when(accountCacheManager.getCurrentAccount()).thenAnswer((_) async => AccountFixtures.aliceAccount);
        
        final response = await dio.post(baseUrl);
        
        expect(response.data, {'message': 'Retry request successfully!'});
      });
```

Reply:
Your writing style is fine too. But I don't think it is the `onGet` function. It will execute that request again with the method call being the previous method.

Reply:
@hoangdat Please see my unit test below